src/Microsoft.Health.Fhir.Core/Features/Conformance/CapabilityStatementBuilder.cs

```diff
@@ -0,0 +1,230 @@
+
+            UpdateRestResourceComponent(resourceType, c =>
+            {
+                foreach (SearchParamComponent searchParam in searchParameters)
+                {
+                    c.SearchParam.Add(searchParam);
+                }
+            });
+
+            return this;
+        }
+
+        public ICapabilityStatementBuilder AddDefaultResourceInteractions()
+        {
+            foreach (var resource in _modelInfoProvider.GetResourceTypeNames())
+            {
+                if (string.Equals(resource, KnownResourceTypes.Parameters, StringComparison.Ordinal))
+                {
+                    continue;
+                }
+
+                TryAddRestInteraction(resource, TypeRestfulInteraction.Create);
+                TryAddRestInteraction(resource, TypeRestfulInteraction.Read);
+
+                if (!string.Equals(resource, KnownResourceTypes.AuditEvent, StringComparison.Ordinal))
+                {
+                    if (!string.Equals(resource, KnownResourceTypes.Binary, StringComparison.Ordinal))
+                    {
+                        TryAddRestInteraction(resource, TypeRestfulInteraction.Vread);
+                    }
+
+                    TryAddRestInteraction(resource, TypeRestfulInteraction.Update);
+                    TryAddRestInteraction(resource, TypeRestfulInteraction.Delete);
+                }
+
+                UpdateRestResourceComponent(resource, component =>
+                {
+                    component.Versioning.Add(ResourceVersionPolicy.NoVersion);
+                    component.Versioning.Add(ResourceVersionPolicy.Versioned);
+                    component.Versioning.Add(ResourceVersionPolicy.VersionedUpdate);
+
+                    component.ReadHistory = true;
+                    component.UpdateCreate = true;
+                });
+            }
+
+            return this;
+        }
+
+        public ICapabilityStatementBuilder AddDefaultSearchParameters()
+        {
+            foreach (var resource in _modelInfoProvider.GetResourceTypeNames())
+            {
+                if (string.Equals(resource, KnownResourceTypes.Parameters, StringComparison.Ordinal))
+                {
+                    continue;
+                }
+
+                IEnumerable<SearchParameterInfo> searchParams = _searchParameterDefinitionManager.GetSearchParameters(resource);
+
+                if (searchParams.Any())
+                {
+                    TryAddSearchParams(resource, searchParams.Select(x => new SearchParamComponent
+                    {
+                        Name = x.Name,
+                        Type = x.Type,
+                        Definition = x.Url,
+                        Documentation = x.Description,
+                    }));
+
+                    TryAddRestInteraction(resource, TypeRestfulInteraction.SearchType);
+                }
+
+                TryAddRestInteraction(resource, TypeRestfulInteraction.HistoryType);
```

Comment:
Should these two be added in the DefaultResourceInteractions above? Doesn't seem to be related to search paramters?

Reply:
That should be fine. History capabilities are tied to search and not the DataStore which is why they originated here.